integration_tests/src/main/python/array_test.py

```diff
@@ -625,3 +625,35 @@ def test_arrays_overlap_before_spark313(data_gen):
             'arrays_overlap(array(3, 4), array(1, 2))',
             'arrays_overlap(array(), array(1, 2))')
     )
+
+@pytest.mark.parametrize('data_gen', [byte_gen, short_gen, int_gen, long_gen], ids=idfn)
+def test_array_remove_scalar(data_gen):
+    gen = StructGen(
+        [('a', ArrayGen(data_gen, nullable=True))],
+        nullable=False)
+
+    assert_gpu_and_cpu_are_equal_collect(
+        lambda spark: gen_df(spark, gen).selectExpr(
+            'array_remove(a, -10)',
+            'array_remove(a, 0)',
+            'array_remove(a, 10)')
+    )
+
+@pytest.mark.parametrize('data_gen', [byte_gen, short_gen, int_gen, long_gen,
+                                      float_gen, double_gen,
+                                      string_gen, boolean_gen, date_gen, timestamp_gen] + decimal_gens, ids=idfn)
+def test_array_remove(data_gen):
+    gen = StructGen(
+        [('a', ArrayGen(data_gen, nullable=True)),
+        ('b', data_gen)],
+        nullable=False)
+
+    assert_gpu_and_cpu_are_equal_collect(
+        lambda spark: gen_df(spark, gen).selectExpr(
+            'array_remove(a, b)',
+            'array_remove(a, null)',
+            'array_remove(array(), null)',
```

Comment:
Just FYI. Spark has an optimization rule where if all of the inputs to an operation are a scalar it will execute the code on the driver on the CPU and replace it with a single scalar value. It is not a big deal, but I just wanted to be sure you know that this is probably not testing what you expect it to be testing.

Reply:
Thank you! TIL :)